### PR TITLE
Doc Hub proposal

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
https://docs.google.com/document/d/1wvCvM6pLQ17Zr0DcpS9es2vOFQsNeOaNN99ufGw8O1U/edit#heading=h.w5wnz0kbiild

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
- Add new entity (descriptionEntity)
- Add description entity to taskSpec and workflowSpec. These two specs are only used in the create request.
- Add GetDescriptionEntity and ListDescriptionEntities endpoints

## Tracking Issue
https://github.com/flyteorg/flyte/issues/531

## Follow-up issue
_NA_

### CODE REVIEW

File: protos/flyteidl/admin/task.proto [hunk 1]
```diff
@@ -51,6 +52,9 @@ message TaskList {
 message TaskSpec {
     // Template of the task that encapsulates all the metadata of the task.
     core.TaskTemplate template = 1;
+
+    // DescriptionEntity encapsulates all the detailed documentation for the task.
+    DescriptionEntity description_entity = 2;
```
Comment (EngHabu) [inline protos/flyteidl/admin/task.proto:57]: I'm a bit confused about this. Perhaps because I'm not sure how this is planned to work out.
There are APIs to explicitly create/retrieve a DescriptionEntity but then the DescriptionEntity is already included in the Admin objects... so which is it? why do we have two paths to do the same thing...

File: protos/flyteidl/admin/description_entity.proto [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+// Link to source code used to define this entity
+message SourceCode {
+  string link = 5;
+}
+
+// Represents a request structure to create a revision of a task.
+// See :ref:`ref_flyteidl.admin.Task` for more details
+message DescriptionEntityCreateRequest {
+  // id represents the unique identifier of the task.
```
Comment (EngHabu) [inline protos/flyteidl/admin/description_entity.proto:57]: nit:
```suggestion
  // id represents the unique identifier of the entity.
```

File: protos/flyteidl/admin/description_entity.proto [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+// Full user description with formatting preserved. This can be rendered
+// by clients, such as the console or command line tools with in-tact
+// formatting.
+message LongDescription {
+  // long description - no more than 4KB
+  string values = 1;
+  // if the description sizes exceed some threshold we can offload the entire
+  // description proto altogether to an external data store, like S3 rather than store inline in the db
+  string uri = 2;
```
Comment (EngHabu) [inline protos/flyteidl/admin/description_entity.proto:34]:
```suggestion
  oneof content {
    // long description - no more than 4KB
    string value = 1;
    // if the description sizes exceed some threshold we can offload the entire
    // description proto altogether to an external data store, like S3 rather than store inline in the db
    string uri = 2;
  }
```

**Review by EngHabu — COMMENTED** [review]
A great start! Should we also "enhance" the parameter descriptions in the same PR? allowing labels... etc.

**Comment by katrogan** [comment]
hey @pingsutw is this proposal ready to be reviewed?

**Comment by pingsutw** [comment]
@katrogan yes, could you take a look, please?
Update: 
- Add `tags` to task/workflow/execution CreateRequest
- Add  `descriptionEntity` to task/workflow/launchPlan, so we can get `entitySpec` and `descriptionEntity` in one RPC call by using getRequest or ListRequest.

File: protos/flyteidl/admin/description_entity.proto [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  }
+  // Format of the long description
+  DescriptionFormat format = 3;
+  // Optional link to an icon for the entity
+  string icon_link = 4;
+}
+
+// Link to source code used to define this entity
+message SourceCode {
+  string link = 5;
```
Comment (katrogan) [inline protos/flyteidl/admin/description_entity.proto:47]:
```suggestion
  string link = 1;
```

File: protos/flyteidl/admin/launch_plan.proto [hunk 5]
```diff
@@ -51,6 +52,9 @@ message LaunchPlan {
 
     // Values computed by the flyte platform after launch plan registration.
     LaunchPlanClosure closure = 3;
+
+    // DescriptionEntity contains detailed description for the launch plan.
+    DescriptionEntity description_entity = 4;
```
Comment (katrogan) [inline protos/flyteidl/admin/launch_plan.proto:57]: I'm worried adding these inline is going to bloat the message size of these flyte entitity payloads by a lot
if necessary we can also add a List endpoint that supports the current semantics for filtering, sorting and pagination?

File: protos/flyteidl/admin/description_entity.proto (hunk 4 above)
Comment (pingsutw) [reply protos/flyteidl/admin/description_entity.proto:47]: I didn't add too much metadata for now in the source code because It's hard to revert it. 
yup, I can add it also if we want it.

Comment (katrogan) [reply protos/flyteidl/admin/description_entity.proto:47]: sounds good, we can iterate on this

File: protos/flyteidl/admin/execution.proto [hunk 6]
```diff
@@ -37,6 +37,10 @@ message ExecutionCreateRequest {
     // included in this map. If not required and not provided, defaults apply.
     // +optional
     core.LiteralMap inputs = 5;
+
+    // User-specified tags. These are arbitrary and can be used for searching
+    // filtering and discovering executions.
+    repeated string tags = 6;
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply protos/flyteidl/admin/execution.proto]: Moved to spec

File: protos/flyteidl/admin/launch_plan.proto (hunk 5 above)
Comment (pingsutw) [reply protos/flyteidl/admin/launch_plan.proto]: sounds reasonable, removed it

File: protos/flyteidl/core/identifier.proto [hunk 7]
```diff
@@ -14,6 +14,7 @@ enum ResourceType {
   // Eventually all Catalog objects should be modeled similar to Flyte Objects. The Dataset entities makes it possible for the UI  and CLI to act on the objects 
   // in a similar manner to other Flyte objects
   DATASET            = 4;
+  DESCRIPTION_ENTITY = 5;
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply protos/flyteidl/core/identifier.proto]: sorry, my bad. we don't need it here. removed it.

File: protos/flyteidl/admin/workflow.proto [hunk 8]
```diff
@@ -55,6 +56,9 @@ message WorkflowSpec {
     // propeller compiler (since the compiler doesn't have any knowledge of other workflows - ie, it doesn't reach out
     // to Admin to see other registered workflows).  In fact, subworkflows do not even need to be registered.
     repeated core.WorkflowTemplate sub_workflows = 2;
+
+    // Represents the specification for Description.
```
Comment (wild-endeavor) [inline protos/flyteidl/admin/workflow.proto:60]: can you add a comment here that logic for listing/getting will masked?  i think we won't return the long description when listing right? only the short description.  Also I think for workflows specifically if they have subworkflows we might mask more stuff too, and users have to call the get description entity to get more.

Comment (pingsutw) [reply protos/flyteidl/admin/workflow.proto:63]: `WorkflowSpec` and `TaskSpec` are used only when creating a workflow and task. Therefore, we send both task/workflow proto and description proto to admin when we call register task/workflow.

I just added a short description to the [task](https://github.com/flyteorg/flyteidl/blob/5b4d98ca4a5280f1cae8b7ba45d91c74add87482/protos/flyteidl/admin/task.proto#L32-L38) and [workflow](https://github.com/flyteorg/flyteidl/blob/5b4d98ca4a5280f1cae8b7ba45d91c74add87482/protos/flyteidl/admin/workflow.proto#L31-L37), and we'll use it when we call get task and list task.

Comment (wild-endeavor) [reply protos/flyteidl/admin/workflow.proto:63]: oh right yes.  thanks.

**Review by wild-endeavor — DISMISSED** [review]

**Review by eapolinario — DISMISSED** [review]

File: protos/flyteidl/admin/description_entity.proto [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+  // In the case of multiple pages of results, the server-provided token can be used to fetch the next page
+  // in a query. If there are no more results, this value will be empty.
+  string token                = 2;
+}
+
+// Represents a request structure to retrieve a list of DescriptionEntities.
+// See :ref:`ref_flyteidl.admin.DescriptionEntity` for more details
+message DescriptionEntityListRequest {
+  // Indicates the node execution to filter by.
```
Comment (EngHabu) [inline protos/flyteidl/admin/description_entity.proto:83]: Can you take a pass on all the field descriptions, some it seems are copy-pasted by mistake?

File: protos/flyteidl/admin/description_entity.proto [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+  // +required
+  DescriptionEntityIdentifier description_entity_id = 1;
```
Comment (EngHabu) [inline protos/flyteidl/admin/description_entity.proto:85]: I don't get why are we passing the identifier in the list request?

File: protos/flyteidl/admin/description_entity.proto [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  SourceCode source_code = 3;
+  // User-specified tags. These are arbitrary and can be used for searching
+  // filtering and discovering tasks.
+  repeated string tags = 4;
+}
+
+// Full user description with formatting preserved. This can be rendered
+// by clients, such as the console or command line tools with in-tact
+// formatting.
+message LongDescription {
```
Comment (EngHabu) [inline protos/flyteidl/admin/description_entity.proto:26]: nit: I think this should be called just `Description`... the field in line #15 called `long_description` is fine...
@wild-endeavor ?

File: protos/flyteidl/admin/description_entity.proto [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+    // if the description sizes exceed some threshold we can offload the entire
+    // description proto altogether to an external data store, like S3 rather than store inline in the db
+    string uri = 2;
+  }
+
+  enum DescriptionFormat {
+    UNKNOWN = 0;
+    MARKDOWN = 1;
+    HTML = 2;
+    // python default documentation - comments is rst
```
Comment (EngHabu) [inline protos/flyteidl/admin/description_entity.proto:39]: Interesting. Will this be a problem for rendering in the Hub? I thought we are expecting markdown.. @wild-endeavor @jsonporter

File: protos/flyteidl/admin/description_entity.proto (hunk 10 above)
Comment (wild-endeavor) [reply protos/flyteidl/admin/description_entity.proto:85]: iiuc - this will filter based on the fields that are filled in in the identifier.  so if you have resource type and project, then it'll do the list on those.  plus whatever extra filters there are.

File: protos/flyteidl/admin/description_entity.proto (hunk 11 above)
Comment (wild-endeavor) [reply protos/flyteidl/admin/description_entity.proto:26]: sure yah

File: protos/flyteidl/admin/description_entity.proto (hunk 12 above)
Comment (wild-endeavor) [reply protos/flyteidl/admin/description_entity.proto:39]: i feel like maybe we can remove this comment actually.  but i don't think it's fair to assume one or the other.  we'll probably have to parse and take a guess.  i'm expecting people to write in both styles, but internally to flyte, all our python comments are written as sphinx rst.

File: protos/flyteidl/admin/description_entity.proto (hunk 10 above)
Comment (wild-endeavor) [reply protos/flyteidl/admin/description_entity.proto:85]: Listing today is done by `admin.ResourceListRequest` or `admin.NamedEntityIdentifierListRequest`.  i think both of these don't differentiate between tasks and workflows, but when listing descriptions we do.  so i think that's why this is here, but yeah @pingsutw can you confirm?

File: protos/flyteidl/admin/description_entity.proto (hunk 12 above)
Comment (EngHabu) [reply protos/flyteidl/admin/description_entity.proto:39]: Mind chatting with Jason about this? Not sure if we will need to translate from one to another or if we can have different renderers... having the enum here is actually useful to determine that behavior

File: protos/flyteidl/admin/description_entity.proto (hunk 10 above)
Comment (EngHabu) [reply protos/flyteidl/admin/description_entity.proto:85]: ic... maybe the name of the message need to change then? as it stands, it's not really an identifier for the description entity.. it's not versioned

Comment (wild-endeavor) [reply protos/flyteidl/admin/description_entity.proto:85]: actually yeah that made me realize... the `DescriptionEntity` object is the only kinda high level user facing object that doesn't carry it's ID with it.  Is that okay?  like if a user calls `GetDescriptionEntity`, the description returned has lost its connection with the root entity.  is this okay?  @EngHabu

Comment (EngHabu) [reply protos/flyteidl/admin/description_entity.proto:85]: `s/it's/its/`

That's a good point... I think maybe we should have a DescriptionEntity wrapper (like those Admin wrappers for tasks), that has the underlying DescriptionEntity object + ID ?

Comment (wild-endeavor) [reply protos/flyteidl/admin/description_entity.proto:85]: (edited)

Comment (wild-endeavor) [reply protos/flyteidl/admin/description_entity.proto:85]: yeah okay, we're going to add id.

File: protos/flyteidl/admin/description_entity.proto [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+package flyteidl.admin;
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin";
+
+import "flyteidl/core/identifier.proto";
+import "flyteidl/admin/common.proto";
+
+// DescriptionEntity contains detailed description for the task/workflow.
+// Documentation could provide insight into the algorithms, business use case, etc.
+message DescriptionEntity {
+  // id represents the unique identifier of the description entity.
```
Comment (wild-endeavor) [inline protos/flyteidl/admin/description_entity.proto:12]: sorry this is the id of the associated task/workflow right?  it's not the id of the description itself.  because the core.Identifier object does not have a resource type for a "description object".

Comment (katrogan) [reply protos/flyteidl/admin/description_entity.proto:12]: any reason not to add an identifier resource type for description entities?

File: protos/flyteidl/admin/task.proto [hunk 14]
```diff
@@ -51,6 +55,9 @@ message TaskList {
 message TaskSpec {
     // Template of the task that encapsulates all the metadata of the task.
     core.TaskTemplate template = 1;
+
+    // Represents the specification for Description.
+    DescriptionEntity description_entity        = 2;
```
Comment (katrogan) [inline protos/flyteidl/admin/task.proto:60]: Is this always going to be populated in requests to Get and List tasks?

Comment (pingsutw) [reply protos/flyteidl/admin/task.proto:60]: TaskSpec is only used for [TaskCreateRequest](https://github.com/flyteorg/flyteidl/blob/cb8df7146b91b8269d1a4ea9b2a1fda42b56b6ff/protos/flyteidl/admin/task.proto#L14-L22), and We use [Task](https://github.com/flyteorg/flyteidl/blob/cb8df7146b91b8269d1a4ea9b2a1fda42b56b6ff/protos/flyteidl/admin/task.proto#L32-L41) for get and list tasks

Comment (katrogan) [reply protos/flyteidl/admin/task.proto:60]: awesome! thanks for explaining

File: protos/flyteidl/admin/description_entity.proto (hunk 9 above)
Comment (pingsutw) [reply protos/flyteidl/admin/description_entity.proto]: sorry, updated it.

File: protos/flyteidl/admin/description_entity.proto (hunk 11 above)
Comment (pingsutw) [reply protos/flyteidl/admin/description_entity.proto]: updated

File: protos/flyteidl/admin/description_entity.proto (hunk 13 above)
Comment (pingsutw) [reply protos/flyteidl/admin/description_entity.proto:12]: > sorry this is the id of the associated task/workflow right

Right. 

We store the description of the task and the workflow in the same table. `resource_type` is one of our primary key in the table, and we use it to look up the specific resource's description. So we didn't add an identifier resource type for description entities.

The table will be like
| resource_type | name | description | ... |
| ------------- | ------------- | ---------- | -------- |
task | t1 | t1 is .... | ... | ...
workflow | wf1 | wf1 is ... | ... 
... | ... | ... | ...

**Review by wild-endeavor — APPROVED** [review]

**Review by wild-endeavor — DISMISSED** [review]

File: protos/flyteidl/service/admin.proto [hunk 15]
```diff
@@ -603,5 +604,26 @@ service AdminService {
       };
   }
 
-}
+  // Fetch a :ref:`ref_flyteidl.admin.DescriptionEntity` object.
+  rpc GetDescriptionEntity (flyteidl.admin.ObjectGetRequest) returns (flyteidl.admin.DescriptionEntity) {
+    option (google.api.http) = {
+      get: "/api/v1/description_entities/{id.project}/{id.domain}/{id.name}/{id.version}"
```
[the remark this answers is not in the document]
Comment (EngHabu) [reply protos/flyteidl/service/admin.proto:610]: Also, `id` refers to a NamedEntityIdentifier which doesn't have a version.. what is this supposed to point to?

File: protos/flyteidl/admin/description_entity.proto [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+message Description {
+  oneof content {
+    // long description - no more than 4KB
+    string value = 1;
+    // if the description sizes exceed some threshold we can offload the entire
+    // description proto altogether to an external data store, like S3 rather than store inline in the db
+    string uri = 2;
+  }
+
+  enum DescriptionFormat {
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply protos/flyteidl/admin/description_entity.proto]: updated it.

File: protos/flyteidl/admin/task.proto (hunk 14 above)
[the remark this answers is not in the document]
Comment (pingsutw) [reply protos/flyteidl/admin/task.proto]: updated it.

File: protos/flyteidl/admin/workflow.proto [hunk 17]
```diff
@@ -55,6 +59,9 @@ message WorkflowSpec {
...
+    DescriptionEntity description_entity        = 3;
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply protos/flyteidl/admin/workflow.proto]: updated it.

File: protos/flyteidl/service/admin.proto (hunk 15 above)
[the remark this answers is not in the document]
Comment (pingsutw) [reply protos/flyteidl/service/admin.proto]: yes, we should add resource_type.

cc @EngHabu one question. I got an error when adding `{id.resource_type}` to the GetDescriptionEntity [endpoint](https://github.com/flyteorg/flyteidl/pull/303/files#diff-9324a86791de0a6631e4fd9acccbe51c4b5f06c33180fd732c69de560e363246R610).
```bash
+ mockery -dir=gen/pb-go/flyteidl/service/ -all -output=clients/go/admin/mocks
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
+ mockery -dir=gen/pb-go/flyteidl/datacatalog/ -name=DataCatalogClient -output=clients/go/datacatalog/mocks
Generating mock for: DataCatalogClient in file: clients/go/datacatalog/mocks/DataCatalogClient.go
go generate ./...
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
```
However, it works if I add {resource_type} to ListDescriptionEntities [endpoint](https://github.com/flyteorg/flyteidl/pull/303/files#diff-9324a86791de0a6631e4fd9acccbe51c4b5f06c33180fd732c69de560e363246R620). I've tried to import `"flyteidl/core/identifier.proto"` in `service/admin.proto`, but it doesn't work. Do I need to import Resource type somewhere.

[the remark this answers is not in the document]
Comment (pingsutw) [reply protos/flyteidl/service/admin.proto]: we use `core.Identifier` in ObjectGetRequest instead of NamedEntityIdentifier. there is a version in `core.Identifier`

**Review by wild-endeavor — DISMISSED** [review]

File: generate_protos.sh [hunk 18]
```diff
@@ -84,6 +85,11 @@ rm -rf gen/pb-go/github.com
 mkdir -p gen/pb_python/validate
 cp -r validate/* gen/pb_python/validate/
 
+# Update the service code manually because the code generated by protoc is incorrect
+# More detail, check https://github.com/flyteorg/flyteidl/pull/303#discussion_r1002151053
+sed -i -e 's/protoReq.Id.ResourceType = ResourceType(e)/protoReq.Id.ResourceType = core.ResourceType(e)/g' gen/pb-go/flyteidl/service/admin.pb.gw.go
+rm -f gen/pb-go/flyteidl/service/admin.pb.gw.go-e
```
Comment (pingsutw) [inline generate_protos.sh:91]: cc @EngHabu Seems like there are some issues in the protoc. I can fix the error by updating it manually.
```bash
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
Error parsing file:  /Users/kevin/git/flyteidl/gen/pb-go/flyteidl/service/admin.pb.gw.go:2743:29: undeclared name: ResourceType
```
Other alternatives:
1. Bump protoc version (May need to update the image `lyft/protocgenerator:8167e11d3b3439373c2f033080a4b550078884a2`)
2. Add `DescriptionEntityGetRequest` in the IDL, which will be like the below. and then update the endpoint to `/api/v1/description_entities/{resource_type}/{id.project}/{id.domain}/{id.name}/{version}`
Note: protoc will fail to generate the code when using `id.resource_type` in the endpoint
```protobuf
message DescriptionEntityListRequest {
  // Identifies the specific type of resource that this identifier corresponds to.
  flyteidl.core.ResourceType resource_type  = 1;
  NamedEntityIdentifier id = 2;
  string version = 3
```

I can remove the change if you don't like it.

Comment (EngHabu) [reply generate_protos.sh:91]: Is there a plan to use the rest endpoint? if not, then maybe let's remove it (the REST endpoint) completely?

Comment (pingsutw) [reply generate_protos.sh:91]: yes, It will be used in flyteconsole

File: protos/flyteidl/core/identifier.proto [hunk 19]
```diff
@@ -19,7 +19,7 @@ enum ResourceType {
 // Encapsulation of fields that uniquely identifies a Flyte resource.
 message Identifier {
     // Identifies the specific type of resource that this identifier corresponds to.
-    ResourceType resource_type  = 1;
+    core.ResourceType resource_type  = 1;
```
Comment (EngHabu) [inline protos/flyteidl/core/identifier.proto:22]: @evalsocket mind taking a look at this? do you have any ideas why this generates bad code?

**Review by wild-endeavor — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]
